pkg/pullsync/pullsync.go

```diff
@@ -174,16 +176,20 @@ func (s *Syncer) SyncInterval(ctx context.Context, peer swarm.Address, bin uint8
 		}
 		s.metrics.Offered.Inc()
 		s.metrics.DbOps.Inc()
-		have, err = s.storage.Has(ctx, a)
-		if err != nil {
-			s.logger.Debug("storage has", "error", err)
-			continue
-		}
-		if !have {
-			wantChunks[a.ByteString()] = struct{}{}
-			ctr++
-			s.metrics.Wanted.Inc()
-			bv.Set(i / swarm.HashSize)
+		po := swarm.Proximity(a.Bytes(), peer.Bytes())
```

Comment:
the proximity calculation should be based on the chunk address and OUR own address, not peer, so you need to pass our overlay address to the pullsync :)

Reply:
the goal of this change is that the chunks we are pulling in must belong in our reserve, then the reserve is chunks with po >= storage radius relative to our address